src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.NumericValueSet.cs

```diff
@@ -48,6 +49,8 @@ internal NumericValueSet(ImmutableArray<(T first, T last)> intervals)
                 }
 #endif
                 _intervals = intervals;
+                _tc = tc;
+                _numericValueSetFactory = new NumericValueSetFactory<T, TTC>(tc);
```

Comment:
>_numericValueSetFactory [](http://example.com/codeflow?start=16&length=23)

Since this is a struct, we should be able to create it on the fly right when it is needed.